### PR TITLE
Using pyodide to generate image in the browser

### PR DESCRIPTION
**Pull request by mliezun**
## Motivation

This project is great. Is sad that I cannot generate images anymore because the backend is not working. Here we update the code to generate images in the frontend so it's free: only github pages is needed.

## Pyodide

[Pyodide](https://pyodide.org/en/stable/) is a port of CPython to WebAssembly. It has interoperability to JavaScript and the browser.

It comes with support for Pillow which is what we need for this project. 

## Work done

We're copying functions from the backend, tweking them a bit to read network instead of local files and inlining the image as base64.

Copied from [backend repo](https://github.com/mislavcimpersak/xkcd-excuse-generator/):
- `app.py`
- `blank_excuse.png`
- `xkcd-script.ttf`

Stripped all the backend code from `app.py`.

Updated `main.js` to call directly to python function: `get_excuse_image(who, why, what)`.

Updated `index.html` to import Pyodide and clean up dead content.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    """
+    errors = []
+
+    errors = _check_user_input_not_empty(errors, who, 1010)
+    errors = _check_user_input_not_empty(errors, why, 1020)
+    errors = _check_user_input_not_empty(errors, what, 1030)
+
+    who = 'The #1  {} excuse'.format(who).upper()
+    legit = 'for legitimately slacking off:'.upper()
+    why = '"{}."'.format(why)
```
Comment (mislavcimpersak) [inline app.py]: Please add `.upper()` to `why` and `what`. In the [original code](https://github.com/mislavcimpersak/xkcd-excuse-generator/blob/master/app.py#L243) a separate function did that, but think we can just do it here.

Example without calling `.upper()`:
image

**Comment by mislavcimpersak** [comment]
This is amazing! It is pure magic 🌟 

Thank you @mliezun so much for working on this!

File: app.py (hunk 1 above)
Comment (mliezun) [reply app.py]: Done! Force pushed the changes so everything is still in one commit

**Comment by mliezun** [comment]
Thank you @mislavcimpersak for building this in the beginning. I hope to be making excuses again soon 😉

**Comment by mislavcimpersak** [comment]
> Thank you @mislavcimpersak for building this in the beginning. I hope to be making excuses again soon 😉

...and we're [back in business](https://xkcd-excuse.com/)! 🚀 

Thank you once again @mliezun 🤗

**Pull request merged**